plugins/modules/network/routeros/routeros_api.py

```diff
@@ -0,0 +1,489 @@
+                elif self.where[1] == '>':
+                    select = self.api_path.select(*keys).where(keys[self.where[0]] > self.where[2])
+                elif self.where[1] == '<':
+                    select = self.api_path.select(*keys).where(keys[self.where[0]] < self.where[2])
+                else:
+                    self.errors("'%s' is not operator for 'where'"
+                                % self.where[1])
+                for row in select:
+                    self.result['message'].append(row)
+            else:
+                for row in self.api_path.select(*keys):
+                    self.result['message'].append(row)
+            if len(self.result['message']) < 1:
+                msg = "no results for '%s 'query' %s" % (' '.join(self.path),
+                                                         ' '.join(self.query))
+                if self.where:
+                    msg = msg + ' WHERE %s' % ' '.join(self.where)
+                self.result['message'].append(msg)
+            self.return_result(False)
+        except Exception as e:
+            self.errors(e)
+
+    def api_arbitrary(self):
+        param = {}
+        self.arbitrary = self.arbitrary.split(' ')
+        arb_cmd = self.arbitrary[0]
+        if len(self.arbitrary) > 1:
+            param = self.list_to_dic(self.arbitrary[1:])
+        try:
+            arbitrary_result = self.api_path(arb_cmd, **param)
+            for i in arbitrary_result:
+                self.result['message'].append(i)
+            self.return_result(False)
+        except Exception as e:
+            self.errors(e)
+
+    def return_result(self, ch_status=False, status=True):
+        if not status:
+            self.module.fail_json(msg=self.result['message'])
+        else:
+            self.module.exit_json(changed=ch_status,
+                                  msg=self.result['message'])
+
+    def errors(self, e):
+        if e.__class__.__name__ == 'TrapError':
+            self.result['message'].append("%s" % e)
+            self.return_result(False, True)
+        self.result['message'].append("%s" % e)
+        self.return_result(False, False)
+
+
+def main():
+    # define available arguments/parameters a user can pass to the module
+    ros = ROS_api_module(dict(
+        username=dict(type='str', required=True, no_log=False),
+        password=dict(type='str', required=True, no_log=True),
+        hostname=dict(type='str', required=True, no_log=False),
+        port=dict(type='int', required=False, no_log=False),
+        ssl=dict(type='bool', required=False, default=False, no_log=False),
+        path=dict(type='str', required=True, no_log=False),
+        add=dict(type='str', required=False, no_log=False),
+        remove=dict(type='str', required=False, no_log=False),
+        update=dict(type='str', required=False, no_log=False),
+        cmd=dict(type='str', required=False, no_log=False),
+        query=dict(type='str', required=False, no_log=False)))
```

Comment:
```suggestion
        query=dict(type='str', required=False, no_log=False),
        ))
```
just a good practice:)

Reply:
1) i think `no_log` is `false` by default but not 100% sure. To get read of unnecessary stuff will make the module clearer.
2) please remove `required=False` because this is the default. the same reason as above

Reply:
Thanks for the review, 

I will try to fix all, however  for no_log=False, I think I do that as related issue to https://github.com/ansible/ansible/issues/49465

Will double check it and will provide more information before I fix it

Reply:
An explicit `no_log=False` should **only** be used when this warning is shown.